eox_nelp/signals/receivers.py

```diff
@@ -227,6 +230,38 @@ def enrollment_publisher(instance, **kwargs):  # pylint: disable=unused-argument
         )
 
 
+def create_usersignupsource_by_enrollment(instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receiver that is connected to the course enrollment post_save signal. This will generate a
+    UserSignupSource record if possilbe
+    based on the `SITE_NAME` tenant settings.
+    If there is not SITE_NAME configurated this would be skipped with a log.
+
+    Args:
+        instance<CourseEnrollment>: This an instance of the model CourseEnrollment.
+    """
+    site_name = TenantSiteConfigProxy.get_value_for_org(instance.course_id.org, "SITE_NAME")
```

Comment:
```suggestion
    site_name = getattr(settings, "SITE_NAME", None)
```
I'm just wondering if this is not the same ? and I worry about the tenants that have multiple orgs

Reply:
@andrey-canon Because you could save a courseEnrollment of lms org using the edunext tenant. So the signUpSource would be attached to the org of the courseEnrollment and not the current tenant request. Also in async could help to don't think in the async tenant.

Reply:
According to tenants having multiple orgs. The think is fetching inverse. The courseEnrollment has a course eg
`course-v1:BRAGI+cd101+2023t1`. That course would match org BRAGI. Then the method would search for a tenantConfig that matches that organization in org_filter(organizations).
After that in the TenanConfig matched, the `SITE_NAME` is retrieved.
The problem could be with multiple tenant_config that have the same org. In this case,
the first one would be selected, according to the following [snippet](https://github.com/eduNEXT/eox-tenant/blob/c4b4b6c716b6b03763c076156b3595ac2c324df2/eox_tenant/models.py#L218-L226).